mobile/src/main/java/org/openhab/habdroid/util/SuggestCommandsFactory.java

```diff
@@ -0,0 +1,149 @@
+package org.openhab.habdroid.util;
+
+import android.content.Context;
+import androidx.annotation.Nullable;
+import androidx.annotation.StringRes;
+
+import org.openhab.habdroid.R;
+import org.openhab.habdroid.model.Item;
+import org.openhab.habdroid.model.LabeledValue;
+import org.openhab.habdroid.model.ParsedState;
+import org.openhab.habdroid.model.Widget;
+
+import java.util.List;
+
+public class SuggestCommandsFactory {
+    Context mContext;
+    List<String> mCommands;
+    List<String> mLabels;
+    boolean mShowUndef;
+
+    public SuggestCommandsFactory(Context context, List<String> commands, List<String> labels,
```

Comment:
I'd prefer to make the lists getters instead of passing them into the constructor. Basically this
```
private final List<String mCommands = new ArrayList<>();
private final List<String mLabels = new ArrayList<>();
[...]
public SuggestCommandsFactory(Context context, boolean showUndef) {
    [...]
}

public List<String> getCommands() {
    return mCommands;
}
public List<String> getLabels() {
    return mLabels;
}
```

(Could maybe even use a single `List<Pair<String, String>> mCommandsAndLabels`)

That way it becomes much clearer from the usage of the class what the class actually does.